src/aliasRequestApiClient.ts

```diff
@@ -0,0 +1,86 @@
+import { IAliasRequest, IAliasCallback } from "./identity.interfaces";
+import { MParticleWebSDK } from "./sdkRuntimeModels";
+import Constants from './constants';
+import { FetchUploader, XHRUploader } from './uploaders';
+
+
+var HTTPCodes = Constants.HTTPCodes,
+    Messages = Constants.Messages;
+
+interface IAliasResponseBody {
+    message?: string
+}
+
+export async function sendAliasRequest (mpInstance: MParticleWebSDK, aliasRequest: IAliasRequest, aliasCallback: IAliasCallback) {
+        const { verbose, error } = mpInstance.Logger;
+        const { invokeAliasCallback } = mpInstance._Helpers;
+        const { aliasUrl } = mpInstance._Store.SDKConfig;
+        const { devToken } = mpInstance._Store;
+
+        verbose(Messages.InformationMessages.SendAliasHttp);
+
+        const uploadUrl = `https://${aliasUrl}${devToken}/Alias`;
+        const uploader = window.fetch
+            ? new FetchUploader(uploadUrl)
+            : new XHRUploader(uploadUrl);
+
+        const uploadPayload = {
+            method: 'post',
+            headers: {
+                Accept: 'text/plain;charset=UTF-8',
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(aliasRequest),
+        };
+        try {
+            const response = await uploader.upload(uploadPayload);
+
+            let message: string;
+            let aliasResponseBody: IAliasResponseBody;
+
+            // FetchUploader returns the response as a JSON object that we have to await
+            if (response.json) {
+                // HTTP responses of 202, 200, and 403 do not have a response.  response.json will always exist on a fetch, but can only be await-ed when the response is not empty, otherwise it will throw an error.
+                try {
+                    aliasResponseBody = await response.json();
+                } catch (e) {
+                    verbose('The request has no response body');
```

Comment:
Shouldn't this be an error?

Reply:
Resolving as we discussed this over zoom